apps/remix-ide/src/remixAppManager.js

```diff
@@ -15,8 +15,8 @@ export function isNative (name) {
   return nativePlugins.includes(name) || requiredModules.includes(name)
 }
 
-export function canActivate (name) {
-  return ['ethdoc'].includes(name) || isNative(name)
+export function canActivate (from, to) {
+  return ['ethdoc'].includes(from.name) || isNative(from.name) || (typeof to !== 'undefined' ? typeof from.canActivate !== 'undefined' ? from.canActivate.includes(to.name) : false : false)
```

Comment:
Could the new test be put as a new line and with maybe a comment. 
Just matter of better readability

Reply:
sure

Reply:
> is `typeof to !== 'undefined' ` really needed?
> same for `typeof from.canActivate`

if you provide a plugin name to call that does not exist to is undefined
canactivate is optional, so it should be checked